marlowe-playground-client/src/SimulationPage/State.purs

```diff
@@ -259,10 +221,7 @@ getPrice exchange pair = do
   pure price
 
 getCurrentContract :: forall m. HalogenM State Action ChildSlots Void m String
-getCurrentContract = do
-  oldContract <- use _oldContract
-  currContract <- editorGetValue
-  pure $ fromMaybe mempty $ oldContract <|> currContract
+getCurrentContract = fromMaybe "Close" <$> editorGetValue
```

Comment:
Yes, you are absolutely right. I left it there because it is only used by the Wallet simulator. But that is no good because someone else could use it. I am going to change this function to return a Maybe, and then I will issue a Console. Warn from the calling function